### PR TITLE
Add libGerminate Event

### PR DESCRIPTION
**Pull request by Brean0**
- removes unneeded stalk burn function.

### CODE REVIEW

File: protocol/contracts/libraries/Silo/LibGerminate.sol [hunk 1]
```diff
@@ -57,6 +57,24 @@ library LibGerminate {
         int256 deltaBdv
     );
 
+    /**
+     * @notice emitted when the total germinating stalk changes. 
+     * @param germinationSeason issuance season of germinating stalk
+     * @param deltaStalk the change in the total germinating stalk.
+     * @dev the issuance season may differ from the season that this function was called in.
```
Comment (soilking) [inline protocol/contracts/libraries/Silo/LibGerminate.sol:64]: suggest "the issuance season may differ from the season that this **event was emitted** in."

File: protocol/contracts/libraries/Silo/LibGerminate.sol [hunk 2]
```diff
@@ -57,6 +57,24 @@ library LibGerminate {
...
+     */
+    event TotalGerminatingStalkChanged(
+        uint256 germinationSeason,
+        int256 deltaStalk
```
Comment (soilking) [inline protocol/contracts/libraries/Silo/LibGerminate.sol:68]: suggest: deltaGerminatingStalk

File: protocol/contracts/libraries/Silo/LibGerminate.sol [hunk 3]
```diff
@@ -129,10 +145,14 @@ library LibGerminate {
...
                 -int256(totalGerm.deposited[tokens[i]].bdv)
             );
-
             // clear deposited values.
             delete totalGerm.deposited[tokens[i]];
         }
+
+        // emit change in total germinating stalk.
+        // safecast not needed as finishedGerminatingStalk is initially a uint128.
+        emit TotalGerminatingStalkChanged(season, -int256(finishedGerminatingStalk));
```
Comment (soilking) [inline protocol/contracts/libraries/Silo/LibGerminate.sol:154]: shouldn't this be `germinationSeason`?

File: protocol/contracts/libraries/Silo/LibSilo.sol [hunk 4]
```diff
@@ -228,42 +228,13 @@ library LibSilo {
...
         }
 
-        // emit event.
-        emit LibGerminate.FarmerGerminatingStalkBalanceChanged(
-            account,
-            stalk
-        );
+        // emit events.
+        emit LibGerminate.FarmerGerminatingStalkBalanceChanged(account, stalk);
+        emit LibGerminate.TotalGerminatingStalkChanged(season, stalk);
```
Comment (soilking) [inline protocol/contracts/libraries/Silo/LibSilo.sol:233]: same here, believe It should be the germinating season

File: protocol/contracts/libraries/Silo/LibSilo.sol [hunk 5]
```diff
@@ -316,10 +287,8 @@ library LibSilo {
         }
 
         // emit events.
-        emit LibGerminate.FarmerGerminatingStalkBalanceChanged(
-            account,
-            -int256(stalk)
-        );
+        emit LibGerminate.FarmerGerminatingStalkBalanceChanged(account, -int256(stalk));
+        emit LibGerminate.TotalGerminatingStalkChanged(season, -int256(stalk));
```
Comment (soilking) [inline protocol/contracts/libraries/Silo/LibSilo.sol:291]: and here

Comment (Brean0) [reply protocol/contracts/libraries/Silo/LibSilo.sol]: fixed here: https://github.com/BeanstalkFarms/Beanstalk/pull/837/commits/cdc5a0353e9298e5d74888fde498d3a9988114cc#diff-fbd431ce899b0679e919b2bf1fa828e50eeddd381e4efd2012eca396bb7758daR295

File: protocol/contracts/libraries/Silo/LibSilo.sol (hunk 4 above)
Comment (Brean0) [reply protocol/contracts/libraries/Silo/LibSilo.sol]: fixed here: https://github.com/BeanstalkFarms/Beanstalk/pull/837/commits/cdc5a0353e9298e5d74888fde498d3a9988114cc#diff-fbd431ce899b0679e919b2bf1fa828e50eeddd381e4efd2012eca396bb7758daR235

File: protocol/contracts/libraries/Silo/LibGerminate.sol (hunk 3 above)
Comment (Brean0) [reply protocol/contracts/libraries/Silo/LibGerminate.sol]: fixed here: https://github.com/BeanstalkFarms/Beanstalk/pull/837/commits/cdc5a0353e9298e5d74888fde498d3a9988114cc#diff-d2261445685829aad9421eae03f9644ae9d11be5713ccc0870497a0be1e58d8aR154

File: protocol/contracts/libraries/Silo/LibGerminate.sol (hunk 2 above)
Comment (Brean0) [reply protocol/contracts/libraries/Silo/LibGerminate.sol]: Fixed here: 
https://github.com/BeanstalkFarms/Beanstalk/pull/837/commits/cdc5a0353e9298e5d74888fde498d3a9988114cc#diff-d2261445685829aad9421eae03f9644ae9d11be5713ccc0870497a0be1e58d8aR68

File: protocol/contracts/libraries/Silo/LibGerminate.sol (hunk 1 above)
Comment (Brean0) [reply protocol/contracts/libraries/Silo/LibGerminate.sol]: Fixed here: https://github.com/BeanstalkFarms/Beanstalk/pull/837/commits/cdc5a0353e9298e5d74888fde498d3a9988114cc#diff-d2261445685829aad9421eae03f9644ae9d11be5713ccc0870497a0be1e58d8aR64

**Pull request merged**